### PR TITLE
Fix `clippy::items_after_statements` lint in struct field alignment tests

### PR DESCRIPTION
**Pull request by onalante-msft**
The bugfix in ae1b386f865e048b1ba2dbea93c621c6a68dfdf1 seems to trigger `clippy::items_after_statements` lints in tests. ~Reordering the field alignment test functions before the invocations resolves this.~ Using a common `MaybeUninit` for all field tests prevents overflowing the stack and removes the need for defining items.

If functions are preferred, a simpler method than that in d6b60d2b6ed5e4671e2dcb90257f96059f02b1cf would be to collect the field alignment tests together with their invocations (to remove the need for two repetitions), but I would imagine having the struct-level tests ordered before the field tests is the semantically more correct option. I am not partial either way, though.

### CODE REVIEW

**Comment by onalante-msft** [comment]
Sorry about re-notifying.

I was intending to hold off on regenerating test expectations until the code generation modifications were approved. However, I can regenerate them now if that is preferred.

**Comment by kulp** [comment]
> However, I can regenerate them now if that is preferred.

It is definitely preferred. Reviewers (including myself) are not as likely to look at PRs that have failing CI.

**Comment by onalante-msft** [comment]
Understood, I will regenerate the tests.

**Review by emilio — APPROVED** [review]
The approach here looks good, thank you!

**Pull request merged**
